src/mpid/ch4/generic/mpidig_send.h

```diff
@@ -14,20 +14,92 @@
 
 #include "ch4_impl.h"
 
+/*
+=== BEGIN_MPI_T_CVAR_INFO_BLOCK ===
+
+cvars:
+    - name        : MPIR_CVAR_CH4_EAGER_MAX_MSG_SIZE
+      category    : CH4
+      type        : int
+      default     : -1
+      class       : device
+      verbosity   : MPI_T_VERBOSITY_USER_BASIC
+      scope       : MPI_T_SCOPE_LOCAL
+      description : >-
+        If set to positive number, this cvar controls the message size at which CH4 switches from eager to rendezvous mode.
+        If the number is negative, underlying netmod or shmmod automatically uses an optimal number depending on
+        the underlying fabric or shared memory architecture.
+
+=== END_MPI_T_CVAR_INFO_BLOCK ===
+*/
+
+#define MPIDIG_AM_SEND_HDR_SIZE  sizeof(MPIDIG_hdr_t)
+
+static inline int mpidig_eager_limit(int is_local)
+{
+    if (MPIR_CVAR_CH4_EAGER_MAX_MSG_SIZE > 0) {
+        return MPIR_CVAR_CH4_EAGER_MAX_MSG_SIZE;
+    }
+
+    int thresh;
+#ifdef MPIDI_CH4_DIRECT_NETMOD
+    thresh = MPIDI_NM_am_eager_limit();
+#else
+    if (is_local) {
+        thresh = MPIDI_SHM_am_eager_limit();
+    } else {
+        thresh = MPIDI_NM_am_eager_limit();
+    }
+#endif
+    thresh -= MPIDIG_AM_SEND_HDR_SIZE;
+    MPIR_Assert(thresh > 0);
+
+    return thresh;
+}
+
+static inline int MPIDIG_do_eager_send(const void *buf, MPI_Aint count, MPI_Datatype datatype,
+                                       int rank, int tag, MPIR_Comm * comm, int context_offset,
+                                       MPIDI_av_entry_t * addr, MPIR_Request ** request,
+                                       MPIR_Errflag_t errflag);
+static inline int MPIDIG_do_rndz_send(const void *buf, MPI_Aint count, MPI_Datatype datatype,
```

Comment:
I think @raffenet already made a comment about naming of this function. Can you use `_rndv_` instead of `_rndz_` for consistency with the rest of the code? Same goes for comments and commit messages.

Reply:
Even though I believe *rndz* is the right abbreviation, I see we have been using *rndz* throughout the existing code base, so I have updated to `MPIDIG_do_rndv_send` instead.